src/compiler/checker.ts

```diff
@@ -9052,11 +9083,10 @@ namespace ts {
         }
 
         function getAliasSymbolForTypeNode(node: TypeNode) {
-            return (node.parent.kind === SyntaxKind.TypeAliasDeclaration || node.parent.kind === SyntaxKind.JSDocTypedefTag) ? getSymbolOfNode(node.parent) : undefined;
+            return isTypeAlias(node.parent) ? getSymbolOfNode(node.parent) : undefined;
         }
 
-        function getAliasTypeArgumentsForTypeNode(node: TypeNode) {
-            const symbol = getAliasSymbolForTypeNode(node);
+        function getTypeParametersForAliasSymbol(symbol: Symbol) {

```

Comment:
This name should contain `TypeArguments` and not `TypeParameters`.

Reply:
I think there is an overall confusion in naming between typeArguments and typeParameters, because this function delegates to getLocalType**Parameters**OfClassOrInstanceOfTypeAlias, which delegates to getEffectiveType**Parameter**Declarations. But then the result of this function is always assigned to aliasType**Arguments**.

Ah, well, a big, complicated change is not the place to drop in another renaming.

Reply:
Type arguments is already definitely correct here - `aliasTypeArguments` always hold the interstated type arguments.